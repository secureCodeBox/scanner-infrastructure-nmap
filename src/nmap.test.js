/*
 *
 *  SecureCodeBox (SCB)
 *  Copyright 2015-2018 iteratec GmbH
 *
 *  Licensed under the Apache License, Version 2.0 (the "License");
 *  you may not use this file except in compliance with the License.
 *  You may obtain a copy of the License at
 *
 *  	http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing, software
 *  distributed under the License is distributed on an "AS IS" BASIS,
 *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *  See the License for the specific language governing permissions and
 *  limitations under the License.
 * /
 */
const { transform, worker } = require('./nmap');
const uuid = require('uuid/v4');
const portscan = require('../lib/portscan');

jest.mock('../lib/portscan');

describe('nmap', () => {
    describe('transform', () => {
        beforeAll(() => {
            jest.mock('uuid/v4');
        });

        beforeEach(() => {
            uuid.mockClear();
        });

        afterAll(() => {
            jest.unmock('uuid/v4');
        });

        it('should transform a empty host array into an empty port array', () => {
            const findings = transform([]);

            expect(uuid).not.toHaveBeenCalled();
            expect(findings).toEqual([]);
        });

        it('should transform a null host array into an empty port array', () => {
            const findings = transform(null);

            expect(uuid).not.toHaveBeenCalled();
            expect(findings).toEqual([]);
        });

        it('should transform results of a single host into a port array', () => {
            const [finding, ...otherFindings] = transform([
                {
                    hostname: 'securebox',
                    ip: '192.168.99.100',
                    mac: null,
                    openPorts: [
                        {
                            port: 22,
                            protocol: 'tcp',
                            service: 'ssh',
                            method: 'table',
                        },
                    ],
                    osNmap: null,
                },
            ]);

            expect(otherFindings).toEqual([]);
            expect(uuid).toHaveBeenCalledTimes(1);
            expect(finding).toEqual({
                id: '49bf7fd3-8512-4d73-a28f-608e493cd726',
                name: 'ssh',
                description: 'Port 22 is open using tcp protocol.',
                category: 'Open Port',
                osi_layer: 'NETWORK',
                severity: 'INFORMATIONAL',
                reference: null,
                hint: null,
                location: 'tcp://192.168.99.100:22',
                attributes: {
                    port: 22,
                    ip_address: '192.168.99.100',
                    protocol: 'tcp',
                    service: 'ssh',
                    method: 'table',
                    hostname: 'securebox',
                    mac_address: null,
                    operating_system: null,
                },
            });
        });

        it('should transform results if a host has multiple open ports', () => {
            const findings = transform([
                {
                    hostname: 'securebox',
                    ip: '192.168.99.100',
                    mac: null,
                    openPorts: [
                        {
                            port: 22,
                            protocol: 'tcp',
                            service: 'ssh',
                            method: 'table',
                        },
                        {
                            port: 80,
                            protocol: 'udp',
                            service: 'http',
                            method: 'table',
                        },
                    ],
                    osNmap: null,
                },
            ]);

            expect(uuid).toHaveBeenCalledTimes(2);
            expect(findings).toEqual([
                {
                    id: '49bf7fd3-8512-4d73-a28f-608e493cd726',
                    name: 'ssh',
                    description: 'Port 22 is open using tcp protocol.',
                    category: 'Open Port',
                    osi_layer: 'NETWORK',
                    severity: 'INFORMATIONAL',
                    reference: null,
                    hint: null,
                    location: 'tcp://192.168.99.100:22',
                    attributes: {
                        port: 22,
                        ip_address: '192.168.99.100',
                        protocol: 'tcp',
                        service: 'ssh',
                        method: 'table',
                        hostname: 'securebox',
                        mac_address: null,
                        operating_system: null,
                    },
                },
                {
                    id: '49bf7fd3-8512-4d73-a28f-608e493cd726',
                    name: 'http',
                    description: 'Port 80 is open using udp protocol.',
                    category: 'Open Port',
                    osi_layer: 'NETWORK',
                    severity: 'INFORMATIONAL',
                    reference: null,
                    hint: null,
                    location: 'udp://192.168.99.100:80',
                    attributes: {
                        port: 80,
                        ip_address: '192.168.99.100',
                        protocol: 'udp',
                        service: 'http',
                        method: 'table',
                        hostname: 'securebox',
                        mac_address: null,
                        operating_system: null,
                    },
                },
            ]);
        });

        it('should transform results of multiple hosts into a port array', () => {
            const findings = transform([
                {
                    hostname: 'securebox',
                    ip: '192.168.99.100',
                    mac: null,
                    openPorts: [
                        {
                            port: 22,
                            protocol: 'tcp',
                            service: 'ssh',
                            method: 'table',
                        },
                    ],
                    osNmap: null,
                },
                {
                    hostname: 'test',
                    ip: '192.168.99.101',
                    mac: null,
                    openPorts: [
                        {
                            port: 80,
                            protocol: 'udp',
                            service: 'http',
                            method: 'table',
                        },
                    ],
                    osNmap: null,
                },
            ]);

            expect(uuid).toHaveBeenCalledTimes(2);
            expect(findings).toEqual([
                {
                    id: '49bf7fd3-8512-4d73-a28f-608e493cd726',
                    name: 'ssh',
                    description: 'Port 22 is open using tcp protocol.',
                    category: 'Open Port',
                    osi_layer: 'NETWORK',
                    severity: 'INFORMATIONAL',
                    reference: null,
                    hint: null,
                    location: 'tcp://192.168.99.100:22',
                    attributes: {
                        port: 22,
                        ip_address: '192.168.99.100',
                        protocol: 'tcp',
                        service: 'ssh',
                        method: 'table',
                        hostname: 'securebox',
                        mac_address: null,
                        operating_system: null,
                    },
                },
                {
                    id: '49bf7fd3-8512-4d73-a28f-608e493cd726',
                    name: 'http',
                    description: 'Port 80 is open using udp protocol.',
                    category: 'Open Port',
                    osi_layer: 'NETWORK',
                    severity: 'INFORMATIONAL',
                    reference: null,
                    hint: null,
                    location: 'udp://192.168.99.101:80',
                    attributes: {
                        port: 80,
                        ip_address: '192.168.99.101',
                        protocol: 'udp',
                        service: 'http',
                        method: 'table',
                        hostname: 'test',
                        mac_address: null,
                        operating_system: null,
                    },
                },
            ]);
        });

        it('should still kind of work if the openPorts attribute of the host is not an array', () => {
            const findings = transform([
                {
                    hostname: 'securebox',
                    ip: '192.168.99.100',
                    mac: null,
                    openPorts: null,
                    osNmap: null,
                },
            ]);

            expect(uuid).not.toHaveBeenCalled();
            expect(findings).toEqual([]);
        });
    });

    describe('worker', () => {
        beforeEach(() => {
            portscan.mockClear();
        });

        it('should work with a single target', async () => {
            const targets = JSON.stringify([{ location: 'localhost' }]);

            const result = await worker({ PROCESS_TARGETS: targets });

            expect(portscan).toBeCalledWith('localhost', '');

            expect(result).toMatchSnapshot();
        });

        it('should take parameters', async () => {
            const targets = JSON.stringify([
                { location: 'localhost', attributes: { NMAP_PARAMETER: '-oX' } },
            ]);

            const result = await worker({ PROCESS_TARGETS: targets });

            expect(portscan).toBeCalledWith('localhost', '-oX');

            expect(result).toMatchSnapshot();
        });

        it('should run multiple scans when multiple targets are configured', async () => {
            const targets = JSON.stringify([
                { location: 'localhost' },
                { location: 'localhost', attributes: { NMAP_PARAMETER: '-oX' } },
            ]);

            const result = await worker({ PROCESS_TARGETS: targets });

            expect(portscan).toHaveBeenCalledTimes(2);
            expect(portscan).toBeCalledWith('localhost', '');
            expect(portscan).toBeCalledWith('localhost', '-oX');

            expect(result).toMatchSnapshot();
        });
    });
});
