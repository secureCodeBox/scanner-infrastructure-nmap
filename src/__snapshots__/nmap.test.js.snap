// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nmap worker should run multiple scans when multiple targets are configured 1`] = `
Object {
  "PROCESS_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"ipp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 631 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":631,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"ipp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:631\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"interwise\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 7778 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":7778,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"interwise\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:7778\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"http-proxy\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 8080 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":8080,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"http-proxy\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:8080\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"wap-wsp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 9200 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":9200,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"wap-wsp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:9200\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"ipp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 631 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":631,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"ipp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:631\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"interwise\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 7778 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":7778,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"interwise\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:7778\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"http-proxy\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 8080 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":8080,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"http-proxy\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:8080\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"wap-wsp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 9200 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":9200,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"wap-wsp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:9200\\\\\\"}]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
  "PROCESS_RAW_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[\\\\\\"<?xml version=\\\\\\\\\\\\\\"1.0\\\\\\\\\\\\\\" encoding=\\\\\\\\\\\\\\"UTF-8\\\\\\\\\\\\\\"?>\\\\\\\\n<!DOCTYPE nmaprun></nmaprun>\\\\\\",\\\\\\"<?xml version=\\\\\\\\\\\\\\"1.0\\\\\\\\\\\\\\" encoding=\\\\\\\\\\\\\\"UTF-8\\\\\\\\\\\\\\"?>\\\\\\\\n<!DOCTYPE nmaprun></nmaprun>\\\\\\"]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
}
`;

exports[`nmap worker should take parameters 1`] = `
Object {
  "PROCESS_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"ipp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 631 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":631,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"ipp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:631\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"interwise\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 7778 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":7778,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"interwise\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:7778\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"http-proxy\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 8080 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":8080,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"http-proxy\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:8080\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"wap-wsp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 9200 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":9200,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"wap-wsp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:9200\\\\\\"}]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
  "PROCESS_RAW_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[\\\\\\"<?xml version=\\\\\\\\\\\\\\"1.0\\\\\\\\\\\\\\" encoding=\\\\\\\\\\\\\\"UTF-8\\\\\\\\\\\\\\"?>\\\\\\\\n<!DOCTYPE nmaprun></nmaprun>\\\\\\"]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
}
`;

exports[`nmap worker should work with a single target 1`] = `
Object {
  "PROCESS_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"ipp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 631 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":631,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"ipp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:631\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"interwise\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 7778 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":7778,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"interwise\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:7778\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"http-proxy\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 8080 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":8080,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"http-proxy\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:8080\\\\\\"},{\\\\\\"id\\\\\\":\\\\\\"49bf7fd3-8512-4d73-a28f-608e493cd726\\\\\\",\\\\\\"name\\\\\\":\\\\\\"wap-wsp\\\\\\",\\\\\\"description\\\\\\":\\\\\\"Port 9200 is open using tcp protocol.\\\\\\",\\\\\\"osi_layer\\\\\\":\\\\\\"NETWORK\\\\\\",\\\\\\"reference\\\\\\":null,\\\\\\"severity\\\\\\":\\\\\\"INFORMATIONAL\\\\\\",\\\\\\"attributes\\\\\\":{\\\\\\"port\\\\\\":9200,\\\\\\"ip_address\\\\\\":\\\\\\"127.0.0.1\\\\\\",\\\\\\"mac_address\\\\\\":null,\\\\\\"protocol\\\\\\":\\\\\\"tcp\\\\\\",\\\\\\"hostname\\\\\\":\\\\\\"localhost\\\\\\",\\\\\\"method\\\\\\":\\\\\\"table\\\\\\",\\\\\\"operating_system\\\\\\":null,\\\\\\"service\\\\\\":\\\\\\"wap-wsp\\\\\\"},\\\\\\"hint\\\\\\":null,\\\\\\"category\\\\\\":\\\\\\"Open Port\\\\\\",\\\\\\"location\\\\\\":\\\\\\"tcp://127.0.0.1:9200\\\\\\"}]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
  "PROCESS_RAW_FINDINGS": ManualSerialization {
    "type": "Object",
    "value": "\\"[\\\\\\"<?xml version=\\\\\\\\\\\\\\"1.0\\\\\\\\\\\\\\" encoding=\\\\\\\\\\\\\\"UTF-8\\\\\\\\\\\\\\"?>\\\\\\\\n<!DOCTYPE nmaprun></nmaprun>\\\\\\"]\\"",
    "valueInfo": Object {
      "objectTypeName": "java.lang.String",
      "serializationDataFormat": "application/json",
    },
  },
}
`;
